### PR TITLE
Partially fixed many-to-many relationship issue

### PR DESCRIPTION
**Pull request by charlesfries**
I think I have a fix for the many-to-many relationship issue people have been experiencing. Please don't merge this in though, as the code is pretty poor right now.  To complete the code we just need a good way of determining if a given relationship's inverse attribute is a belongsTo or a hasMany. Currently I am doing that by checking if the last character is an 's' or not, which obviously is less than ideal. But it does seem to work.

TLDR: This code will work in a pinch if your models follow standard inflection (e.g. "users/user", but not "people/person").

### CODE REVIEW

**Comment by jamesdaniels** [comment]
I fortunately have a better way to detect the inverse relationship. 

What about saving the associated _ids?

**Comment by charlesfries** [comment]
Is this what you mean by saving the associated IDs? The code below is what I'm using to save models with a many-to-many rel:

```
this.store.createRecord('club', { title: 'My Club' }).save().then(club => {
  let member = this.store.createRecord('member', { name: 'Bob' });
  member.clubs.pushObject(club);
  member.save().then(savedMember => {
    club.members.pushObject(savedMember);
    club.save();
  });
});
```

This requires saving the club model twice. I don't know 100% how Ember Data intended relationship fetching/saving to work, but are you implying that the associated IDs should be persisted automatically if a many-to-many relationship is detected?

**Comment by charlesfries** [comment]
@googlebot I signed it!

**Comment by jamesdaniels** [comment]
I think we can save the inverse record in a batch. I’ll look into this more, thanks for getting this kicked off.

**Comment by Alonski** [comment]
@jamesdaniels Any plans to continue this work please?

**Comment by mohitsud** [comment]
@charlesfries is your solution tested enough that i can roll a custom fork and begin to leverage many to many relationships again?
